packages/oslo-validator-jsonld/lib/JsonldValidationService.ts

```diff
@@ -167,65 +166,138 @@ export class JsonldValidationService implements IService {
       invalidEntries: [],
     };
 
-    // Find all quads with apDefinition, vocDefinition, apUsageNote, vocUsageNote predicate
-    const apDefinitionPredicate = ns.oslo('apDefinition');
-    const vocDefinitionPredicate = ns.oslo('vocDefinition');
-    const apUsageNotePredicate = ns.oslo('apUsageNote');
-    const vocUsageNotePredicate = ns.oslo('vocUsageNote');
-    let quads: any[] = [
-      ...this.store.findQuads(null, apDefinitionPredicate, null),
-      ...this.store.findQuads(null, vocDefinitionPredicate, null),
-      ...this.store.findQuads(null, apUsageNotePredicate, null),
-      ...this.store.findQuads(null, vocUsageNotePredicate, null),
+    // Find all quads with apLabel and vocLabel to get all entities
+    const apLabelPredicate = ns.oslo('apLabel');
+    const vocLabelPredicate = ns.oslo('vocLabel');
+
+    let labelQuads: RDF.Quad[] = [
+      ...this.store.findQuads(null, apLabelPredicate, null),
+      ...this.store.findQuads(null, vocLabelPredicate, null),
     ];
 
-    for (const quad of quads) {
-      if (quad.object.termType === 'Literal') {
-        const uri: string = quad.subject.value;
-        const value: string = quad.object.value;
+    // Get unique subjects from label quads
+    const uniqueSubjects = new Set(
+      labelQuads.map((quad) => quad.subject.value)
+    );
 
-        if (this.checkIsEmpty(value)) {
-          this.logger.warn(`Found empty sentence for subject: ${uri}`);
-          result.invalidEntries.push({
-            uri,
-            location: `Sentences may not be empty strings: ${value}`,
-          });
-          continue;
-        }
+    for (const subjectValue of uniqueSubjects) {
+      const subject = labelQuads.find(
+        (quad) => quad.subject.value === subjectValue
+      )?.subject;
+      if (!subject) continue;
+
+      // Check what types of labels exist for this subject
+      const hasApLabel =
+        this.store.findQuads(subject, ns.oslo('apLabel'), null).length > 0;
```

Comment:
This should be exactly one or we have problems later. In RDF you can have multiple Quads but we can only set one label in EA.